games-roguelike/FTL-gog/FTL-gog-1.6.3.ebuild

```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit desktop unpacker
+
+DESCRIPTION="Top-down roguelike space ship simulator"
+HOMEPAGE="https://www.gog.com/game/faster_than_light"
+SRC_URI="ftl_advanced_edition_en_1_6_3_17917.sh"
+RESTRICT="bindist fetch mirror strip test"
+
+LICENSE="GOG-EULA"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+IUSE=""
+
+DEPEND="app-arch/unzip"
+RDEPEND="media-libs/freetype
+	media-libs/libsdl[X,sound,opengl,video]
+	media-libs/libpng
+	sys-libs/zlib
+	virtual/opengl"
+
+S="${WORKDIR}/data/noarch"
+
+pkg_nofetch() {
+	elog "Please buy and download ${SRC_URI} from"
+	elog "https://www.gog.com/game/faster_than_light"
+	elog "and copy it to ${DISTDIR}"
+}
+
+src_unpack() {
+	unpack_zip "${DISTDIR}/${A}"
```

Comment:
Any reason you're using this special function instead of the default `unpack` function that doesn't require any eclasses?

Reply:
Yes, there is a reason. The payload is a shell script that is prefixed to a binary payload, which is really just a ZIP archive. If you notice some comments above from an earlier discussion, I had originally written a call to dd to separate the shell script portion from the ZIP archive, since `unzip` will throw a warning about the unnecessary bytes leading up to the actual binary portion, and it was suggested to simply use this unpack call which doesn't seem to concern itself with the small aforementioned infringement.

Reply:
Apparently the only difference is that it passes `-q` to silence the output. Looks like a really big cannon to silence a harmless informational output (and at the same time lose any valuable output there could be).